maven-docgen/src/main/java/org/apache/maven/tools/CollectConfiguration.java

```diff
@@ -0,0 +1,369 @@
+
+    private static String getConfigurationType(JavaDocCapable<?> javaDocCapable) {
+        String type = getTag(javaDocCapable, "@configurationType");
+        if (type != null) {
+            String linkPrefix = "{@link ";
+            String linkSuffix = "}";
+            if (type.startsWith(linkPrefix) && type.endsWith(linkSuffix)) {
+                type = type.substring(linkPrefix.length(), type.length() - linkSuffix.length());
+            }
+            String javaLangPackage = "java.lang.";
+            if (type.startsWith(javaLangPackage)) {
+                type = type.substring(javaLangPackage.length());
+            }
+        }
+        return nvl(type, "n/a");
+    }
+
+    private static String getConfigurationSource(JavaDocCapable<?> javaDocCapable) {
+        String source = getTag(javaDocCapable, "@configurationSource");
+        if ("{@link RepositorySystemSession#getConfigProperties()}".equals(source)) {
+            return "Session Configuration";
+        } else if ("{@link System#getProperty(String,String)}".equals(source)) {
+            return "Java System Properties";
+        } else if ("{@link org.apache.maven.api.model.Model#getProperties()}".equals(source)) {
+            return "Model Properties";
+        } else if ("{@link Session#getUserProperties()}".equals(source)) {
+            return "Session Properties";
+        } else {
+            return source;
+        }
+    }
+
+    private static String getSince(JavaDocCapable<?> javaDocCapable) {
+        List<JavaDocTag> tags;
+        if (javaDocCapable != null) {
+            if (javaDocCapable instanceof FieldSource<?> fieldSource) {
+                tags = fieldSource.getJavaDoc().getTags("@since");
+                if (tags.isEmpty()) {
+                    return getSince(fieldSource.getOrigin());
+                } else {
+                    return tags.get(0).getValue();
+                }
+            } else if (javaDocCapable instanceof JavaClassSource classSource) {
+                tags = classSource.getJavaDoc().getTags("@since");
+                if (!tags.isEmpty()) {
+                    return tags.get(0).getValue();
+                }
+            }
+        }
+        return "";
+    }
+
+    private static String getTag(JavaDocCapable<?> javaDocCapable, String tagName) {
+        List<JavaDocTag> tags;
+        if (javaDocCapable != null) {
+            if (javaDocCapable instanceof FieldSource<?> fieldSource) {
+                tags = fieldSource.getJavaDoc().getTags(tagName);
+                if (tags.isEmpty()) {
+                    return getTag(fieldSource.getOrigin(), tagName);
+                } else {
+                    return tags.get(0).getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    private static final Pattern CONSTANT_PATTERN = Pattern.compile(".*static final.* ([A-Z_]+) = (.*);");
+
+    private static final ToolProvider JAVAP = ToolProvider.findFirst("javap").orElseThrow();
```

Comment:
Is `javap` always expected to exist?

Reply:
For allowed Java versions to build Maven, yes. And if it get deprecated in some future Java we will know.